public/common/components/match-renderer/view-models/match-renderer.viewmodel.coffee

```diff
@@ -0,0 +1,35 @@
+class MatchRendererViewModel
+	constructor: ({node_list, match}) ->
+		tags = @tags = new nx.Cell
+			'<-': [
+				node_list,
+				(nodes) ->
+					if nodes
+						nodes.map ({type, element, attrs, text, objective}) ->
```

Comment:
The mapping function could be extracted into a ViewModel which would store the node and add the `match` cell